chat.go

```diff
@@ -43,12 +54,70 @@ type ChatCompletionRequest struct {
 	FrequencyPenalty float32                 `json:"frequency_penalty,omitempty"`
 	LogitBias        map[string]int          `json:"logit_bias,omitempty"`
 	User             string                  `json:"user,omitempty"`
+	Functions        []*FunctionDefine       `json:"functions,omitempty"`
```

Comment:
any reason to make `Functions` a slice of pointer? It makes constructing request harder. Why not `[]FunctionDefine`?